newlib/libc/string/strerror_s.c

```diff
@@ -0,0 +1,272 @@
+            ".lib section in a.out corrupted",
+            "Attempting to link in too many libs",
+            "Attempting to exec a shared library",
+            "Function not implemented",
+            "No more files",
+            "Directory not empty",       /* 90 */
+            "File or path name too long",
+            "Too many symbolic links",
+            "unknown error",
+            "unknown error",
+            "Operation not supported on socket",
+            "Protocol family not supported",
+            "unknown error",
+            "unknown error",
+            "unknown error",
+            "unknown error",       /* 100 */
+            "unknown error",
+            "unknown error",
+            "unknown error", 
+            "Connection reset by peer",
+            "No buffer space available",
+            "Address family not supported by protocol family",
+            "Protocol wrong type for socket",
+            "Socket operation on non-socket",
+            "Protocol not available",
+            "Can't send after socket shutdown",       /* 110 */
+            "Connection refused",
+            "Address already in use",
+            "Software caused connection abort",
+            "Network is unreachable",
+            "Network interface is not configured",
+            "Connection timed out",
+            "Host is down",
+            "Host is unreachable",
+            "Connection already in progress",
+            "Socket already connected",       /* 120 */
+            "Destination address required",
+            "Message too long",
+            "Unknown protocol",
+            "Socket type not supported",
+            "Address not available",
+            "Connection aborted by network",
+            "Socket is already connected",
+            "Socket is not connected",
+            "Too many references: cannot splice",
+            "Too many processes",       /* 130 */
+            "Too many users",
+            "Reserved",
+            "Reserved",
+            "Not supported",
+            "No medium found",
+            "unknown error",
+            "unknown error",
+            "Illegal byte sequence",
+            "Value too large for defined data type",
+            "Operation canceled",       /* 140 */
+            "State not recoverable",
+            "Previous owner died",
+            "Streams pipe error",
+            "Memory page has hardware error",
+            "Is a named type file",
+            "Key has expired",
+            "Key was rejected by service",
+            "Key has been revoked",
+            NULL
+    };
+
+    const char *cp;
+    if ((uint32_t)errnum  < SYS_NERR)
+    {
+        cp = SYS_ERRLIST(errnum);
+    }
+    else
+    {
+        /* Standard says any value of type int shall be mapped to a message */
+        cp = "unknown error";
+    }
+    return cp;
+}
+
+/* C11 version; required by LLVM's C++11 library */
+errno_t strerror_s(char *buf, rsize_t buflen, errno_t errnum)
```

Comment:
Let's use the existing strerror mechanism (yeah, it's pretty clunky) instead of replicating the strings here.